cmake/flatbuffers.cmake

```
@@ -0,0 +1,17 @@
+# Function to generate FlatBuffers C++ headers from schema files
+function(GENERATE_FBS_HEADER FBS_FILE)
+    get_filename_component(FBS_FILE_NAME ${FBS_FILE} NAME)
```

For a given input: `foo/bar.h`

Do you wish your generated file to be `flatbuffers/bar.h_generated.h` or `flatbuffers/bar_generated.h`?

For the latter, use this instead:

```suggestion
    get_filename_component(FBS_FILE_NAME ${FBS_FILE} NAME_WE)
```

---

Actually I see in the command to `flatc` we aren't specifying what the final filename is; only its location.  So this MUST match whatever `flatc` is doing, else it will forever be out of date and regenerate over and over and over.

Which filename does `flatc` produce?

---

Looks okay to me, with my existing changes past couple months, they are generated here:

```
build/tt_metal/impl/flatbuffers/program_types_generated.h
build/tt_metal/impl/flatbuffers/buffer_types_generated.h
build/tt_metal/impl/flatbuffers/binary_generated.h
build/tt_metal/impl/flatbuffers/command_generated.h
build/tt_metal/impl/flatbuffers/base_types_generated.h
```

included in my cpp code using:

```
#include "command_generated.h"
#include "binary_generated.h"
```

(I used to have it as this, but suggestion was to make include path more narrow).

```
#include "flatbuffers/command_generated.h"
#include "flatbuffers/binary_generated.h"
```

---

Oh man, I see what you mean now... you're absolutely right. A typo in my helper function here all along (should have used NAME_WE) so dependencies weren't tracked properly causing the headers to be regenerated each time despite no changes. Fixed.